### PR TITLE
HEEDLS-355: Show multiple validation errors in text input VC

### PR DESCRIPTION
**Pull request by SteveJacksonSoft**
@stellake The registration create-password page doesn't use the VC, so it still only shows one validation error per input. Do you want me to change that so it uses the VC as well?

### CODE REVIEW

**Comment by SteveJacksonSoft** [comment]
@stellake How does this look?
image

File: DigitalLearningSolutions.Web/ViewModels/Common/TextInputViewModel.cs [hunk 1]
```diff
@@ -26,8 +29,8 @@ public TextInputViewModel
             SpellCheck = spellCheck;
             Autocomplete = autocomplete;
             HintText = hintText;
-            ErrorMessage = errorMessage;
-            HasError = hasError;
+            ErrorMessages = errorMessageList;
+            HasError = errorMessageList.Any(message => !string.IsNullOrWhiteSpace(message));
```
Comment (stellake) [inline DigitalLearningSolutions.Web/ViewModels/Common/TextInputViewModel.cs:33]: Just out of curiosity - is this consistent with the ErrorSummary component?

**Comment by stellake** [comment]
> @stellake How does this look?
> image

This looks perfect! 👍 

Also, as discussed on standup, David will handle the registration page (just so it doesn't look like I'm ignoring you hehe). :)

File: DigitalLearningSolutions.Web/ViewModels/Common/TextInputViewModel.cs (hunk 1 above)
Comment (SteveJacksonSoft) [reply DigitalLearningSolutions.Web/ViewModels/Common/TextInputViewModel.cs:33]: Ah, it's not actually.

Comment (SteveJacksonSoft) [reply DigitalLearningSolutions.Web/ViewModels/Common/TextInputViewModel.cs:33]: Do you want me to remove the predicate from the Any(), or add it to ErrorSummary?

Comment (stellake) [reply DigitalLearningSolutions.Web/ViewModels/Common/TextInputViewModel.cs:33]: I think it's almost impossible that we'll add a whitespace or null error in the code, so either is fine as long as we make sure that it's consistent across the two components :)

File: DigitalLearningSolutions.Web/Styles/viewComponents/textInput.scss [hunk 2]
```diff
@@ -0,0 +1,8 @@
+.nhsuk-error-message.validation-errors-container__message {
+  margin-bottom: 4px;
+}
+
+.validation-errors-container {
+  padding-top: 4px;
+  padding-bottom: 16px;
+}
```
Comment (stellake) [inline DigitalLearningSolutions.Web/Styles/viewComponents/textInput.scss:8]: Do you think any of the nhsuk-spacing values would work here? https://service-manual.nhs.uk/design-system/styles/spacing

e.g.
margin-bottom: nhsuk-spacing(1);

**Review by stellake — APPROVED** [review]
Looks good - just a question whether we can use nhsuk-spacing values in CSS, but no need to rereview. 👍

**Pull request merged**